src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,412 @@
+
+    # remove any trailing slashes from user provided output_dir
+    output_dir = os.path.normpath(output_dir)
+
+    # make directory for models results
+    model_results_dir = f"{output_dir}/{model_name}"
+    os.makedirs(model_results_dir, exist_ok=True)
+
+    return f"{model_results_dir}/results_{timestamp}.{file_format}"
+
+
+def print_results(
+    result: EvaluationResult, results_files: List[str], model_name: str
+) -> None:
+    """
+    Prints a scoring report for DK-Bench
+    Args:
+        result (Evaluation):       ragas EvaluationResult to parse for scores
+                                   to each question.
+        results_files (List[str]): List of files with scores and responses
+        model_name (str):          Name of model that generated responses for
+                                   DK-Bench to evaluate againist reference answer.
+    Returns:
+        None
+    """
+    print("\n")
+    print("# DK-BENCH REPORT")
+    print(f"\n## MODEL: {model_name}\n")
+    total_score = 0
+    for i, score in enumerate(result.scores):
+        print(f"Question #{i+1}:     {score['domain_specific_rubrics']}/5")
+        total_score += score["domain_specific_rubrics"]
+
+    average = total_score / len(result.scores)
+    print("----------------------------")
+    print(f"Average Score:   {average:.2f}/5")
+    print(f"Total Score:     {total_score}/{len(result.scores)*5}\n")
+
+    print("Responses and scores written to:")
+    for file in results_files:
+        print(f"{file}")
+    print("\n")
+
+
+def create_excel_results_file(excel_file: str, result: EvaluationResult) -> None:
+    """
+    Writes an excel file based on the result of an evaluation.
+    The excel files has two sheets. The first is a summary sheet
+    with a table of individual question scores, average, total score and median.
+
+    The second sheet has the score, the question (user_input), reference,
+    response, model name, and evaluation run, similar to the contents of the
+    .jsonl and .csv files.
+
+    Args:
+        result (Evaluation):    ragas EvaluationResult to parse for scores
+                                for summary sheet and questions, references,
+                                and responses sheet.
+        excel_file (str):       Name of excel file to be created the summary
+                                and dataset sheets.
+    Returns:
+        None
+    """
+    scores = [score["domain_specific_rubrics"] for score in result.scores]
+    question_indices = [f"Q{i + 1}" for i in range(len(scores))]
+
+    col1 = ["Average", "Total Score", "Median", "Question"] + question_indices
+    col2 = [stats.mean(scores), sum(scores), stats.median(scores), "Score"] + scores
+
+    summary_data = {
+        "Metric": col1,
+        "Value": col2,
+    }
+
+    summary_df = pd.DataFrame(summary_data)
+
+    response_df = result.dataset.to_pandas()
+    response_df["scores"] = scores
+
+    with pd.ExcelWriter(excel_file) as writer:
```

Comment:
We need to explicitly select `openpyxl` here in order to prevent the file from being corrupted.


```suggestion
    with pd.ExcelWriter(excel_file, engine="openpyxl") as writer:
```